tests/pthread/main_thread_wait.cpp

```diff
@@ -18,22 +18,6 @@ int main() {
     return 0;
   }
 
-  int x = EM_ASM_INT({
-    onerror = function(e) {
-      var message = e.toString();
-      var success = message.indexOf("Blocking on the main thread is not allowed by default. See https://emscripten.org/docs/porting/pthreads.html#blocking-on-the-main-browser-thread") >= 0;
```

Comment:
why are these tests changed? they lose the property of testing the error message on this line.

Reply:
If you look at the corresponding change to the `test_pthread_main_thread_blocking` test you can see that we can test this directly in the `expected=` argument to `btest` now.

Much cleaner, no needs for weird EM_ASM.

Reply:
Oh wow, nice!